terminal/command.go

```diff
@@ -408,18 +421,37 @@ func printcontext(state *api.DebuggerState) error {
 	}
 
 	if len(state.CurrentThread.File) == 0 {
-		fmt.Printf("Stopped at: 0x%x\n", state.CurrentThread.PC)
+		fmt.Printf("Stopped at: 0x%x\n", state.CurrentThread.Addr)
 		fmt.Printf("\033[34m=>\033[0m    no source available\n")
 		return nil
 	}
 
-	var context []string
-
 	fn := ""
 	if state.CurrentThread.Function != nil {
 		fn = state.CurrentThread.Function.Name
 	}
-	fmt.Printf("current loc: %s %s:%d\n", fn, state.CurrentThread.File, state.CurrentThread.Line)
+
+	// Here because of a command other than 'list' (e.g. 'next')
+	if state.List.Line == 0 {
+		state.List.Line = state.CurrentThread.Line
+		fmt.Printf("current loc: %s %s:%d\n", fn,
+			state.CurrentThread.File, state.CurrentThread.Line)
+	}
+
+	numLines := state.List.ListSize
+	halfLines := (numLines + 1) / 2
+
+	if state.List.Line < halfLines {
+		state.List.Line = halfLines
+	}
+	state.List.FirstLine = state.List.Line - halfLines + 1
+	state.List.LastLine = state.List.Line + halfLines
+
+	return print(state)
+}
+
+func print(state *api.DebuggerState) error {
```

Comment:
I think it'd be great to have this be a method off of `Line` instead of as a standalone function that returns a formatted string.